libheif/box.cc

```diff
@@ -2854,7 +2854,9 @@ std::string Box_idat::dump(Indent& indent) const
   std::ostringstream sstr;
   sstr << Box::dump(indent);
 
-  sstr << indent << "number of data bytes: " << get_box_size() - get_header_size() << "\n";
+  if (get_box_size() > get_header_size()) {
+    sstr << indent << "number of data bytes: " << get_box_size() - get_header_size() << "\n";
+  }
```

Comment:
Could we add something like
````
else {
  sstr << indent << "number of data bytes is invalid\n";
}
````

Reply:
Maybe `>=` ? Size 0 might be a valid edge case.

Reply:
> Maybe `>=` ? Size 0 might be a valid edge case.

I checked 14496-12:2022 Section 8.11.11 and didn't see any prohibition, so assume that 0 is valid. Have updated the code.

Reply:
Have updated the PR to include these.